pkg/multimodelconfig/configmap.go

```diff
@@ -0,0 +1,112 @@
+package multimodelconfig
+
+import (
+	"encoding/json"
+	"k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/runtime/log"
+)
+
+var logger = log.Log.WithName("MultiModelConfig")
+
+type ConfigsDelta struct {
+	updatedCfg map[string]string
```

Comment:
Why use map[string]string instead of the objects themselves?
```
type ModelConfigDelta struct {
  updated []ModelConfig
  delete []ModelCnofig
}
```